File: Prime/Fortran/Prime.f
      program Prime
      implicit none
      integer i,n,Func,f
  100 write (*, '(a)', advance='no') 'number : '
      read(*,*) n
      if(n<0) then
      	n=0
      end if
      
      f = Func(n)
      if (f .eq. 0) then
          write (*,*) 'Composite number'
      else if (f . eq. 1) then
          write (*,*) 'Prime number'
      else
          write (*,*) 'Error'
      end if
      write (*,*)
      write (*, '(a)', advance='no') 'Continue? (Yes : 1 / No : 0) : '
      read(*,*) i
      
      write (*,*)
      
      if (i .ne. 0) then
          goto 100
      end if
      
      stop
      end program
      
      function Func(x)
      implicit none
      integer x,i,Func
      if (x .lt. 2) then
          Func = -1
      else if (x .eq. 2) then
          Func = 1
      else if (mod(x,2) .eq. 0) then
          Func = 0
      else
          Func = 1
          i = 3
          do 200 i=3,int(sqrt(real(x))),2
              if (mod(x,i) .eq. 0) then
                  Func = 0
                  goto 200
              end if
  200     continue
      end if
      end function

File: Prime/Java/Prime.java
import java.util.Scanner;
public class Prime {
	private static int Prime(int x) {
	    if (x < 2) return -1;
	    if (x == 2) return 1;
	    if (x%2 == 0) return 0;
	    for (int i = 3;i <= (int)Math.sqrt(x);i+=2) {
	        if (x % i == 0) return 0;
	    }
	    return 1;
	}
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str,output[]={"Error","Composite number","Prime number"};
        int i,n;
        do {
            System.out.print("number : ");
            str = sc.nextLine();
            try {
        	    n = Integer.parseInt(str);
            }
            catch(NumberFormatException e) {
        	    n = 0;
            }
            if (n < 0) {
        	    n = 0;
            }
            System.out.println("");
            System.out.println(output[Prime(n)+1]);
            System.out.print("Continue? (Yes : 1 / No : 0) : ");
            str = sc.nextLine();
            try {
        	    i = Integer.parseInt(str);
            }
            catch(NumberFormatException e) {
        	    i = 0;
            }
            System.out.println("");
        } while (i != 0);
    }
}